bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/split/strategy/PartitionDivideSplitConstructor.java

```diff
@@ -0,0 +1,197 @@
+import org.apache.kudu.client.Partition;
+import org.apache.kudu.client.PartitionSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PartitionDivideSplitConstructor extends AbstractKuduSplitConstructor {
+  private static final Logger LOG = LoggerFactory.getLogger(PartitionDivideSplitConstructor.class);
+
+  private SplitConfiguration splitConf = null;
+  private boolean available = false;
+  private transient List<Partition> partitions = null;
+
+  public PartitionDivideSplitConstructor(BitSailConfiguration jobConf, KuduClient client) throws Exception {
+    super(jobConf, client);
+
+    if (!jobConf.fieldExists(KuduReaderOptions.SPLIT_CONFIGURATION)) {
+      LOG.warn("{} cannot work due to lack of split configuration.", this.getClass().getSimpleName());
+      return;
+    }
+    String splitConfStr = jobConf.get(KuduReaderOptions.SPLIT_CONFIGURATION);
+    this.splitConf = new ObjectMapper().readValue(splitConfStr, SplitConfiguration.class);
+    KuduTable kuduTable = client.openTable(this.tableName);
+    this.available = splitConf.isValid(kuduTable);
+    if (!available) {
+      LOG.warn("{} cannot work because split configuration is invalid.", this.getClass().getSimpleName());
+      return;
+    }
+    this.available = fillSplitConf(jobConf, client);
+    if (!available) {
+      return;
+    }
+    LOG.info("Successfully created ,final split configuration is: {}", splitConf);
+  }
+
+  @Override
+  public boolean isAvailable() {
+    return available;
+  }
+
+  @Override
+  @SuppressWarnings("checkstyle:MagicNumber")
+  protected boolean fillSplitConf(BitSailConfiguration jobConf, KuduClient client) throws Exception {
+    if (splitConf.getSplitNum() == null || splitConf.getSplitNum() <= 0) {
+      splitConf.setSplitNum(jobConf.getUnNecessaryOption(KuduReaderOptions.READER_PARALLELISM_NUM, 1));
+    }
+    KuduTable kuduTable = client.openTable(this.tableName);
+    this.partitions = kuduTable.getRangePartitions(splitConf.getTimeOut());
+    if (this.partitions.size() < splitConf.getSplitNum()) {
+      this.splitConf.setSplitNum(this.partitions.size());
+      LOG.info("Resize split num to {}.", splitConf.getSplitNum());
+    }
+    if (splitConf.timeOut == null || splitConf.timeOut <= 0) {
+      splitConf.setTimeOut(3000L);
+    }
+    return true;
+  }
+
+  @Override
+  public List <KuduSourceSplit> construct(KuduClient kuduClient) throws Exception {
+    List<KuduSourceSplit> splits = new ArrayList <>();
+    ColumnSchema partitionColumn = schema.getColumn(splitConf.getPartitionKey());
+
+    for (int i = 0; i < partitions.size(); i++) {
+      KuduSourceSplit split = new KuduSourceSplit(i);
+      split.addPredicate(KuduPredicate.newComparisonPredicate(
+          partitionColumn,
+          KuduPredicate.ComparisonOp.GREATER_EQUAL,
+          partitions.get(i).getPartitionKeyStart()[splitConf.partitionKeyIndex]));
```

Comment:
getPartitionKeyStart() returns a byte array.
Why do we need to get one single byte from the array?

Reply:
public static KuduPredicate newComparisonPredicate(ColumnSchema column,
                                                     ComparisonOp op,
                                                     long value)

Reply:
Because I thought  this method   `KuduPredicate.newComparisonPredicate` can just support one ColumnSchema so we can just support one column in `SplitConfiguration` , Now I think I am wrong。

Reply:
> getPartitionKeyStart() returns a byte array. Why do we need to get one single byte from the array?
I thought `KuduPredicate.newComparisonPredicate` just support one ColumnSchema and we can just support one column in `SplitConfiguration`, I  found I was wrong.   
Now I think it is unnecessary for the user to specify one column in `SplitConfiguration`, and we can define the split directly according to all the columns returned by the `Schema.primaryKeyColumns()` method and the values ​​returned by `getRangeKeyStart()` and `getRangeKeyEnd()` because the table  should be simply mode  and do not have  hashBucketSchema.  What do you think?  Is it correct ? @BlockLiu

Reply:
I fixed the code according to my idea above, please review again, thx  @BlockLiu